src/boot/efi/boot.c

```diff
@@ -110,14 +110,18 @@ typedef struct {
  *
  * The other values may be set by systemd-boot itself and changing those will lead to functional regression
  * when new version of systemd-boot is installed.
+ *
+ * All the 64bit values are not ABI and will never be written to an efi variable.
  */
 enum {
-        TIMEOUT_MIN         = 1,
-        TIMEOUT_MAX         = UINT32_MAX - 2U,
-        TIMEOUT_UNSET       = UINT32_MAX - 1U,
-        TIMEOUT_MENU_FORCE  = UINT32_MAX,
-        TIMEOUT_MENU_HIDDEN = 0,
-        TIMEOUT_TYPE_MAX    = UINT32_MAX,
+        TIMEOUT_MIN           = 1,
+        TIMEOUT_MAX           = UINT32_MAX - 2U,
+        TIMEOUT_UNSET         = UINT32_MAX - 1U,
+        TIMEOUT_MENU_FORCE    = UINT32_MAX,
+        TIMEOUT_MENU_HIDDEN   = 0,
+        TIMEOUT_TYPE_MAX      = UINT32_MAX,
+        TIMEOUT_MENU_DISABLED = (uint64_t)UINT32_MAX + 1U,
+        TIMEOUT_TYPE_MAX64    = UINT64_MAX,
```

Comment:
This is a mess. I believe this was shoe-horned into u32 due to gnu-efi short comings. So let's clean this up and drop these weird `TYPE_MAX` constants and also, move the comment inline:

```
        TIMEOUT_MENU_HIDDEN   = 0,
        TIMEOUT_MIN           = 1,
        TIMEOUT_MAX           = UINT32_MAX,
       /* Following values are not ABI and will never be written to an efi variable. */
        TIMEOUT_UNSET = (uint64_t)UINT32_MAX + 1U,
        TIMEOUT_MENU_DISABLED,
        TIMEOUT_MENU_FORCE,
```

Reply:
Please enlighten me to the use case for a u32:MAX - 4 timeout. I'd sure like to know.

Reply:
> I think you've missed the comment just above. Doing this kind of change would be neat, but has a huge chance of causing regressions in obscure ways.

These people just get the default timeout value (or the cap). And anyone who's bothered can easily fix it up in the config or with bootctl. It's not like it renders their system unbootable.

> More importantly, unwrapping these is practically orthogonal to what I'm aiming here.

Well, you're touching this enum and it's getting even more unwieldy, so not really.

Reply:
You know what, go f yourself: #29597

Reply:
please, let's knock it down a notch or two, there's no need